docs/cloud/data-ingestion.md

```diff
@@ -21,3 +21,104 @@ If your Graphite stack does not currently contain any relay, then you can simply
+
+## Sending a copy of your carbon traffic to GrafanaCloud
+
+It is possible to duplicate your data to send one copy to your existing Graphite infrastructure and the other to GrafanaCloud. To do this you can put an instance of carbon-relay-ng in front of your existing carbon-relay or carbon-cache and make it duplicate the traffic. Carbon-relay-ng allows you to specify routes of various types, to send a copy to GrafanaCloud you need to add a route of the type `grafanaNet`, to send a copy to your existing carbon-relay/carbon-cache you can add a carbon route.
+
+For example if you currently have carbon-relay listening on port `2003` and all of your infrastructure is sending its carbon traffic there, you could change it to listen on port `2053` instead and then start a carbon-relay-ng on port `2003` with this config to send a copy of the traffic to `localhost:2053`:
+
+```
+[[route]]
+key = 'carbon'
+type = 'sendAllMatch'
+destinations = [
+    'localhost:2053 spool=true pickle=false'
+]
+
+[[route]]
+key = 'grafanaNet'
+type = 'grafanaNet'
+addr = 'https://<Your endpoint address>'
+apikey = '<Your Grafana.com API Key>'
+schemasFile = '/etc/carbon-relay-ng/storage-schemas.conf'
+```
+
+# High availability and scaling of carbon-relay-ng
+
+## Scaling with carbon-relay-ng
+
+When distributing traffic among multiple instances of carbon-relay-ng it is important to ensure that the same metrics always get sent to the same carbon-relay-ng instances to preserve the order of the data points, this can be done using consistent hashing.
+The carbon-relay and carbon-relay-ng daemons both support consistent hashing to distribute traffic among its carbon destinations. If the carbon-relay(-ng) instance which does the consistent hashing itself becomes a bottleneck it is also ok to have multiple carbon-relay(-ng) instances running in parallel and doing the consistent hashing, among these instances the traffic may then be round-robed.
+
+Example which uses carbon-relay-ng to do the consistent hashing before forwarding the carbon traffic to multiple other carbon-relay-ng instances:
+```
+            incoming carbon traffic
+                       |
+                 <round-robin>
+                /             \
+|-------------------|    |-------------------|
+| carbon-relay-ng-1 |    | carbon-relay-ng-2 |
+|-------------------|    |-------------------|
+             |                     |
+  <consistent hashing>  <consistent hashing>
+             |      \    /         |
+             |       \  /          |
+             |        \/           |
+             |        /\           |
+             |       /  \          |
+             |      /    \         |
+|-------------------|    |-------------------|
+| carbon-relay-ng-3 |    | carbon-relay-ng-4 |
+|-------------------|    |-------------------|
+                    \    /
+               |--------------|
+               | GrafanaCloud |
+               |--------------|
+```
```

Comment:
That's some beautiful ASCII art, lol

Reply:
So just to clarify, all 4 carbon-relay-ng instances are on local (not grafana cloud) hardware.
 2 of them are dedicated hashes, the other two are just there to forward to grafana cloud?

Reply:
lol thx. yes, this setup would completely be running on the customer side. realistically, most customers do not seem to require that, but we did receive the questions of how to scale across instances and how to achieve high availability quite a few times so i just wanted to explain how one -can- do that if they really want to.

Reply:
btw i just realized that this doesn't work, because at the round robin stage the metric data would lose the order, which is a problem. i'll need to fix that in the graph

Reply:
what is the status of this?

Reply:
removed it, there is no round robin anymore